online-store/src/app/app.ts

```diff
@@ -0,0 +1,278 @@
+    }
+}
+
+function filtered () {
+    let newArr: ProductDescription[] = sortArray;
+
+    if (filtersObj.country?.length) {
+        newArr = newArr.filter(el => filtersObj.country?.includes(el.country));
+    }
+    if (filtersObj.brand?.length) {
+        newArr = newArr.filter(el => filtersObj.brand?.includes(el.brand));
+    }
+    if (filtersObj.price.length) {
+        newArr = newArr.filter(el => el.price >= filtersObj.price[0]);
+    }
+    if (filtersObj.price.length) {
+        newArr = newArr.filter(el => el.price <= filtersObj.price[1]);
+    }
+    createTable(newArr);
+
+
+    function sortered () {
+        createTable(sortingArray (newArr, filtersObj.sorting));
+        const selectSort = document.querySelector('.select-box') as HTMLInputElement;
+        if (selectSort) {
+            selectSort.onchange = function() {
+                filtersObj.sorting = selectSort.value;
+                createTable(sortingArray (newArr, filtersObj.sorting));
+            };
+        }
+    }
+    sortered ();
+
+    function sumofChange (arr: ProductDescription[]) {
+        const presentValue: NodeListOf<Element> = document.querySelectorAll('.checkbox-label');
+        presentValue.forEach (el => {
+            const valElem = el.childNodes[0] as HTMLInputElement;
+            let sum = 0;
+                arr.forEach(el => {
+                    if (Object.values(el).includes(valElem.value)) {
+                        sum++;
+                    }
+                });
+            let checkboxValue = el.childNodes[2].childNodes[0] as HTMLInputElement
+            checkboxValue.innerText = `${sum}`;
+        });
+    }
+    sumofChange(newArr);
+
+    function search() {
+        let searchInput: HTMLInputElement | null = document.querySelector('.header-input');
+        let searchArr;
+        if (searchInput){
+            searchInput.oninput = function (){
+                let valThis = this as HTMLInputElement;
+                let val = valThis.value.trim();
+                console.log (val);
+                searchArr = newArr.filter((el) => {
+                    if (el.country.toLowerCase().includes(val.toLowerCase()) || 
+                        el.brand.toLowerCase().includes(val.toLowerCase()) || 
+                        el.title.toLowerCase().includes(val.toLowerCase()) || 
+                        el.surface.toLowerCase().includes(val.toLowerCase()) || 
+                        el.drawing.toLowerCase().includes(val.toLowerCase())) {
+                        return true;
+                    }  else {
+                        return false;
+                    }
+                })
+                createTable(searchArr);
+                sumofChange (searchArr);
+            }
+        }
+    
+        let inputText = document.querySelector('.header-button') as HTMLInputElement;
+        inputText?.addEventListener('click', e => {
+            search();
+        })
+        
+    }
+    search();
+    return newArr;
+}
```

Comment:
- Naming: The function name "filtered" is not very descriptive and does not clearly convey what the function does. It would be better to name it something like "applyFilters".
- 
- Sort and Filter Order: The sorting should be done after all the filtering is done, so it would be better to move the call to sortingArray to the end of the function.
- 
- Check for Empty Inputs: You should check if the filtersObj properties country, brand, and price are not null or undefined before accessing their properties. You can use a conditional (if (filtersObj.country && filtersObj.country.length)) to do this.
- 
- Filtering on Price: Currently, the code only filters products based on the lower bound of the price range. You should also filter based on the upper bound of the price range.
- 
- Repeated Code: The code to check the length of filtersObj.price is repeated twice, but it should only be done once. You can move this check to the beginning of the function and reuse the result.
- 
- Function Reusability: The sumofChange and search functions could be moved outside of the filtered function and made into separate, reusable functions.
- 
- Typing: The type of selectSort can be more specific. Instead of using HTMLInputElement, you can use HTMLSelectElement.